src/java_bytecode/java_static_initializers.cpp

```diff
@@ -114,6 +114,7 @@ static void create_clinit_wrapper_symbols(
   symbolt wrapper_method_symbol;
   code_typet wrapper_method_type;
   wrapper_method_type.return_type() = void_typet();
+  wrapper_method_type.parameters(); // to ensure the property is there
```

Comment:
Why is this necessary?! The only impact that I can think of is the hash value and comparison between irepts.

Reply:
Exactly for that reason.

Reply:
So what code is it that is impacted by changes in the ordering resulting from those differences here?

Reply:
goto-diff

Reply:
But then the same program should always yield the same `irept` (previously without an `ID_parameters` in `named_sub` and now with such an entry). Hence I'm still surprised that this impacts goto-diff in some way.

Reply:
In principle, yes. The root cause is likely an instability elsewhere in the code. I found this issue only after diffing two 200MB irep dumps. We'll have to investigate this further by shrinking the large example that I have. Any similar sources of nondeterminism will cause further trouble in future. For now, this fix will solve the problem for the customer.

Reply:
You might just want to log this in an issue to ensure immediate progress while not forgetting about the problem. As the issue surfaces in goto-diff, however, which data structure in goto-diff is it that is being traversed and found to be spuriously different?